### PR TITLE
Update root README, add infra README, follow infra/gcp/bash mv

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of umbrella issue to refactor infra/gcp: https://github.com/kubernetes/k8s.io/issues/516
- Followup to infra/gcp/bash move: https://github.com/kubernetes/k8s.io/pull/2632

The first few commits fix dead links or references caused by https://github.com/kubernetes/k8s.io/pull/2632

The next two are README updates:
- Add infra/gcp/README.md based in part on https://github.com/kubernetes/k8s.io/issues/516#issuecomment-859948218
- Update /README.md to reflect current apps or paths of interest

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,22 +13,22 @@ Kubernetes project infrastructure, managed by the kubernetes comunity via [wg-k8
...
 - `groups`: google groups on the kubernetes.io domain
 - `hack`: scripts used for development, testing, etc.
 - `images`: container images published to `gcr.io/k8s-staging-infra-tools`
 - `infra/gcp`: scripts and data to manage our GCP infrastructure
-    - `clusters/modules`: terraform modules intended for re-use within this repo
-    - `clusters/projects`: terraform to manage (parts of) GCP projects
+    - `bash/namespaces`: scripts and data to manage K8s namespaces and RBAC for `aaa`
+    - `bash/roles`: scripts and data to manage custom GCP IAM roles
+    - `terraform/modules`: terraform modules intended for re-use within this repo
+    - `terraform/projects`: terraform to manage (parts of) GCP projects
```
Comment (ameukam) [inline README.md:24]:
```suggestion
    - `bash/namespaces`: scripts and data to manage K8s namespaces and RBAC for `aaa`
    - `bash/prow`: scripts and data used to manage projects used for e2e testing and managed by boskos
    - `bash/roles`: scripts and data to manage custom GCP IAM roles
    - `terraform/modules`: terraform modules intended for re-use within this repo
    - `terraform/projects`: terraform to manage (parts of) GCP projects
```

File: infra/gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+## bash
+
+### deploying changes
+
+The bash in here needs to be manually run by humans. At the moment, this is
+usually restricted to members of k8s-infra-gcp-org-admins@kubernetes.io
+
+### principles
```
Comment (ameukam) [inline infra/gcp/README.md:45]: @spiffxp my life goal is to reach your level of communication skills. :-)

File: infra/gcp/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```yaml
+infra:
+  {project_type}: # ideally this ~= folder
+    [managed_by:] # default script/file used to manage projects
+    projects:
+      {project_id}:
+        [managed_by:] overrides default; script used to manage _this_ projec
+```
+
+Please refer to [`infra.yaml`](/infra/gcp/infra.yaml) to see which `ensure-*.sh`
```
Comment (spiffxp) [inline infra/gcp/README.md:105]: Dead link (or we should move infra.yaml here and update bash accordingly)

File: infra/gcp/README.md (hunk 2 above)
Comment (spiffxp) [reply infra/gcp/README.md:45]: I'll leave the README here for now then

**Pull request merged**